aquadoggo/src/materializer/tasks/dependency.rs

```diff
@@ -302,4 +366,161 @@ mod tests {
             .unwrap()
             .unwrap();
     }
+
+    // Helper that creates a schema field definition view in the store.
+    async fn insert_schema_field_definition_view(
+        db: &TestSqlStore,
+        context: &Context,
+    ) -> OperationId {
+        let author = Author::try_from(db.key_pairs[0].public_key().to_owned()).unwrap();
+        let fields = operation_fields(vec![
+            ("name", OperationValue::Text("field_name".to_string())),
+            ("type", FieldType::String.into()),
+        ]);
+        let operation_id = random_operation_id();
+        let op = verified_operation(
+            Some(fields),
+            None,
+            Some(SchemaId::SchemaFieldDefinition(1)),
+            Some(author),
+            Some(operation_id.clone()),
+        );
+        let document_id = DocumentId::new(operation_id.clone());
+        let document_view_id = DocumentViewId::from(operation_id.clone());
+
+        db.store.insert_operation(&op, &document_id).await.unwrap();
+
+        let input = TaskInput::new(None, Some(document_view_id));
+        reduce_task(context.clone(), input.clone()).await.unwrap();
+        operation_id
+    }
+
+    // Helper that creates a schema definition view in the store.
+    async fn insert_schema_definition_view(
+        field_view_ids: Vec<DocumentViewId>,
+        db: &TestSqlStore,
+        context: &Context,
+    ) -> OperationId {
+        let author = Author::try_from(db.key_pairs[0].public_key().to_owned()).unwrap();
+        let fields = operation_fields(vec![
+            ("name", OperationValue::Text("schema_name".to_string())),
+            (
+                "description",
+                OperationValue::Text("description".to_string()),
+            ),
+            (
+                "fields",
+                OperationValue::PinnedRelationList(PinnedRelationList::new(field_view_ids)),
+            ),
+        ]);
+        let operation_id = random_operation_id();
+        let op = verified_operation(
+            Some(fields),
+            None,
+            Some(SchemaId::SchemaDefinition(1)),
+            Some(author),
+            Some(operation_id.clone()),
+        );
+        let document_id = DocumentId::new(operation_id.clone());
+        let document_view_id = DocumentViewId::from(operation_id.clone());
+
+        db.store.insert_operation(&op, &document_id).await.unwrap();
+
+        let input = TaskInput::new(None, Some(document_view_id));
+        reduce_task(context.clone(), input.clone()).await.unwrap();
+        operation_id
+    }
+
+    #[rstest]
+    #[tokio::test]
+    async fn dispatches_schema_tasks_for_field_definitions(
+        #[from(test_db)]
+        #[with(1, 1)]
```

Comment:
Also `insert_entry_operation_and_view()` in `db/stores/test_utils` could be useful i think.

Reply:
Instead of doing the above inline insertion of the entry and operation, you could of course just call `insert_entry_operation_and_view()` once for the `schema_field_definition` and once for the `schema_definition` whichever you prefer.

Reply:
I spent some time trying to refactor with these hints and it is really beautiful for the first test! But then the second test needs a schema definition document that references a schema field definition document and constructing that relation with these decorators seems impossible to me. Because of that I think I will need the two helper functions for creating these documents anyway. And then I might as well use them for the first test as well so that the two tests use a similar structure :(

Reply:
I had a go at refactoring them, let me know what you think! Sure there are other similar approaches too.